### PR TITLE
fix: standardize output path

### PR DESCRIPTION
**Pull request by bcai2**
* Refactors `output_path` to refer _only_ to a directory. If a tool needs a file name specified as well, it should be passed into the `output_primary_name` parameter.
* Removes redundant output variables (`output_name`) and all logic related to whether `output_path` is a directory.
* Adds the `output_path` parameter in `Output.download()`. This should be used in place of the `output_dir` parameter, though the `output_dir` parameter is still supported for backwards compatibility.

This is a **breaking change**. The function calls for the following tools are affected:
* `clustalo`
* `diamond_blastp`
* `diamond_blastx`
* `rapsearch2`

For these tools, `output_path` formerly referred to an output **file**. The base file name should now be parsed into `output_primary_name` instead, and `output_path` should contain the directory the output file will be in.

### CODE REVIEW

File: toolchest_client/tools/tool.py [hunk 1]
```diff
@@ -297,7 +279,6 @@ def _handle_termination(self, signal_number, *_):
         if self.terminating:
             raise InterruptedError("Toolchest client force killed")
         self.terminating = True
-        sentry_sdk.set_tag('send_page', False)
```
Comment (lebovic) [inline toolchest_client/tools/tool.py:300]: Is this removal intentional?

**Review by lebovic — APPROVED** [review]
LGTM

File: toolchest_client/tools/tool.py (hunk 1 above)
Comment (bcai2) [reply toolchest_client/tools/tool.py:300]: Not as part of this PR. I'm guessing that this resulted from a merge from staging, but I'll investigate further

Comment (bcai2) [reply toolchest_client/tools/tool.py:300]: Looks like this was from a merge commit from staging into this branch.
